File: app/src/main/java/eu/napcode/gonoteit/di/modules/UtilsModule.java
package eu.napcode.gonoteit.di.modules;

import dagger.Module;
import dagger.Provides;
import eu.napcode.gonoteit.ui.login.UserValidator;

@Module
public class UtilsModule {

    @Provides
    UserValidator providesUserValidator() {
        return new UserValidator();
    }
}

File: app/src/main/java/eu/napcode/gonoteit/di/modules/UtilsModule.kt
package eu.napcode.gonoteit.di.modules

import dagger.Module
import dagger.Provides
import eu.napcode.gonoteit.auth.StoreAuth
import eu.napcode.gonoteit.utils.ApiClientProvider
import javax.inject.Singleton

@Module
class UtilsModule {

    @Provides
    @Singleton
    internal fun provideApiProvider(storeAuth: StoreAuth): ApiClientProvider {
        return ApiClientProvider(storeAuth)
    }
}
